kernel/src/snapshot.rs

```diff
@@ -81,12 +80,14 @@ impl LogSegment {
         let schema = get_log_schema().project(&[PROTOCOL_NAME, METADATA_NAME])?;
         // filter out log files that do not contain metadata or protocol information
         use Expression as Expr;
-        let meta_predicate = Some(Expr::or(
-            Expr::not(Expr::is_null(Expr::column("metaData.id"))),
-            Expr::not(Expr::is_null(Expr::column("protocol.minReaderVersion"))),
-        ));
+        lazy_static::lazy_static!(
```

Comment:
Meta-predicates don't change, so we can define them as lazy static arc. And then just shallow-clone the arc as needed.

(again below)

Reply:
We should probably use `OnceLock` now that it's in std.

https://doc.rust-lang.org/stable/std/sync/struct.OnceLock.html